src/js/handlebarsHelpers/canonicalGenerator.js

```diff
@@ -0,0 +1 @@
+module.exports = (siteUrl, pageUrl) => siteUrl + pageUrl.replace('index', '')
```

Comment:
Does this deal with trailing slashes as well?  For example `/events/`, `/events`, and `/events/index.html` should all render the correct page and have a canonical url of `BASE_URL/events`

May be worth looking at this: https://github.com/HyunSeob/hexo-auto-canonical/blob/master/index.js

Reply:
@theycallmeswift no, but I don't think we need to. 

Unlike Jekyll where it decides what page url is, here it's passed in to match the name of the html file. So pageUrl will always be 'about' or 'projects' or whatever the html file is called.

Same with siteUrl it's defined by developer in config.

Seeing how we have full control over the two strings, I can't think of anything else to do for this function for now.